### DISCUSSION TITLE
Thread safe maps

### DISCUSSION THREAD

**Discussion by medvednikov** [vlang/v#11729] [Ideas]

**Comment by ntrel** [comment]
Would this be a different type from non-thread safe map? We don't want to lock unnecessarily. Maybe there's a way to still share the implementation code.

**Comment by medvednikov** [comment]
Same way as with @UweKrueger 's `shared foo []int` implementation.

**Comment by UweKrueger** [comment]
Yes, I've been wanting to implement this for a long time...
